### PR TITLE
Remove all dependencies

### PR DESCRIPTION
**Pull request by EnriqueVidal**
The idea is to remove all dependencies like ruby and instead make use of [GNU/stow](https://www.gnu.org/software/stow) to link all existing config files.

This should make it easier to keep local changes in sync with repo (for maintainers) and eliminate the need for installing  rbenv, homebrew etc before being able to produce the files.

- [x] Eliminate dependencies and assume software is installed or will be installed by user
- [x] Replace Vundle with vim-plug
- [x] Cleanup installer to use `stow` and `curl` instead
- [x] Remove Vagrant and add Dockerfile to see the dotfiles (most of them anyways) installed
- [x] Update README to refllect changes

## Trying out

To quickly test use the `Dockerfile` in this branch like so:

```sh
docker build . -t dotfiles
docker run --rm -it dotfiles bash
```

Once inside the container you have many options:

```sh
# Install from local files
cd dotfiles
./install -s
```

```sh
# Fetch from git and install
./dotfiles/install -b features/stow
```

```sh
# Use curl to install from git
curl https://raw.githubusercontent.com/cloverinteractive/dotfiles/features/stow/install | bash -s -- -b features/stow
```

Then load newly linked dotfiles `source ~/.bash_profile`.

**Note**: The first options is  obviously the fastest as it will use your branch local files, the other two is for testing things over the network, keep in mind not using any arguments will fetch from master which may differ greatly from your local branch.

**Note**: If you're installing from curl make sure that the url of the installer matches this branch as well as sending the `-b` argument with a matching branch.

**Note**: If you're installing locally by skipping clone make sure you cd into the folder where the configs exist as the installer will asume it's sitting at the repo's folder.

Edit: Closes #70

### CODE REVIEW

File: X/.i3/config [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This file has been auto-generated by i3-config-wizard(1).
+# This file has been auto-generated by i1-config-wizard(1).
```
Comment (aalvarado) [inline X/.i3/config:1]: `i1` ?

**Comment by aalvarado** [comment]
Check if you like https://github.com/jeffmhubbard/multilockscreen better than betterlockscreen

File: X/.i3/config (hunk 1 above)
Comment (EnriqueVidal) [reply X/.i3/config:1]: heh, vim... must have lowered that on accident, will patch

**Comment by EnriqueVidal** [comment]
> Check if you like https://github.com/jeffmhubbard/multilockscreen better than betterlockscreen

They look pretty similar, why don't we finish removing all the deps and we start a separate PR for that?

**Pull request merged**